rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -605,11 +552,58 @@ def find_elements_by(how, what, parent = nil)
           ids.map { |id| Element.new self, element_id_from(id) }
         end
 
+        protected
+
+        def capabilities_class
+          W3CCapabilities
+        end
+
+        def default_capabilities
+          capabilities_class.firefox
+        end
+
         private
 
-        def assert_javascript_enabled
-          return if capabilities.javascript_enabled?
-          raise Error::UnsupportedOperationError, 'underlying webdriver instance does not support javascript'
+        def bridge_module
+          Module.nesting[1]
+        end
+
+        def process_capabilities(opts)
+          capabilities = opts.delete(:desired_capabilities) || default_capabilities
+          capabilities = Capabilities.send(capabilities) if capabilities.is_a?(Symbol)
```

Comment:
Ok, so the use case is that someone does this:
```
caps = Selenium::WebDriver::Remote::Capabilities.firefox
driver = Selenium::WebDriver.for :remote, desired_capabilities: caps
```
Right now it is using defaults for `Selenium::WebDriver::Remote::Capabilities` instead of defaults for `Selenium::WebDriver::Remote::W3CCapabilities`. I'll go ahead and add this to the top of `Selenium::WebDriver::Remote::Capabilities#self.firefox`:
```
return Remote::W3CCapabilities.firefox(opts) if Remote::W3CCapabilities.w3c?
```
That should address the user-facing concerns I think.